lib/private/Files/Storage/Common.php

```diff
@@ -898,6 +899,11 @@ public function writeStream(string $path, $stream, int $size = null): int {
 
 	public function getDirectoryContent($directory): \Traversable {
 		$dh = $this->opendir($directory);
```

Comment:
Question: 
```suggestion
        try {
		    $dh = $this->opendir($directory); 
		} catch(\Exception $e) {
		    // Throw and take other actions?
		}
```

Can't be caught? To capture the actual failure/error? (That might required an update to method `opendir` if the reading functions through meaningful error messages.

Reply:
If that method throws the exception should already propagate, don't think we need to wrap that.

The local storage implementation would already log an error but return false which just wasn't handled before.